pkg/pool-stable-phantom/test/StablePoolAmplification.test.ts

```diff
@@ -0,0 +1,300 @@
+import { expect } from 'chai';
+import { ethers } from 'hardhat';
+import { BigNumber, Contract } from 'ethers';
+
+import { sharedBeforeEach } from '@balancer-labs/v2-common/sharedBeforeEach';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address';
+import { bn } from '@balancer-labs/v2-helpers/src/numbers';
+import {
+  advanceTime,
+  advanceToTimestamp,
+  currentTimestamp,
+  DAY,
+  setNextBlockTimestamp,
+} from '@balancer-labs/v2-helpers/src/time';
+import * as expectEvent from '@balancer-labs/v2-helpers/src/test/expectEvent';
+import { deploy } from '@balancer-labs/v2-helpers/src/contract';
+import Vault from '@balancer-labs/v2-helpers/src/models/vault/Vault';
+
+describe('StablePoolAmplification', () => {
+  let owner: SignerWithAddress, admin: SignerWithAddress, other: SignerWithAddress;
+  let pool: Contract;
+
+  const AMP_PRECISION = 1e3;
+  const AMPLIFICATION_PARAMETER = bn(200);
+
+  sharedBeforeEach('setup signers', async () => {
+    [, admin, owner, other] = await ethers.getSigners();
+  });
+
+  sharedBeforeEach('deploy pool', async () => {
+    const vault = await Vault.create({ admin });
+    pool = await deploy('MockStablePoolAmplification', {
+      args: [vault.address, owner.address, AMPLIFICATION_PARAMETER],
+    });
+  });
+
+  describe('startAmplificationParameterUpdate', () => {
+    context('when the sender is allowed', () => {
+      context('when requesting a reasonable change duration', () => {
+        const duration = DAY * 2;
+        let endTime: BigNumber;
+
+        sharedBeforeEach('set end time', async () => {
+          const startTime = (await currentTimestamp()).add(100);
+          await setNextBlockTimestamp(startTime);
+          endTime = startTime.add(duration);
+        });
+
+        context('when requesting a valid amp', () => {
+          const itUpdatesAmpCorrectly = (newAmp: BigNumber) => {
+            const increasing = AMPLIFICATION_PARAMETER.lt(newAmp);
+
+            context('when there is no ongoing update', () => {
+              it('starts changing the amp', async () => {
+                await pool.connect(owner).startAmplificationParameterUpdate(newAmp, endTime);
+
+                await advanceTime(duration / 2);
```

Comment:
Can we advance to any point other than the midpoint? :upside_down_face: e.g. one third